src/Data/List/NonEmpty/Base.agda

```diff
@@ -143,6 +143,24 @@ concatMap f = concat ∘′ map f
 ap : List⁺ (A → B) → List⁺ A → List⁺ B
 ap fs as = concatMap (λ f → map f as) fs
 
+-- Inits
+
+inits : List A → List⁺ (List A)
+inits xs = [] ∷ go xs
+  where
+  go : List A → List (List A)
+  go []       = []
+  go (x ∷ xs) = List.[ x ] ∷ List.map (x ∷_) (go xs)
+
+-- Tails
+
+tails : List A → List⁺ (List A)
+tails xs = xs ∷ go xs
+  where
+  go : List A → List (List A)
+  go []       = []
+  go (x ∷ xs) = xs ∷ go xs
```

Comment:
Ah, so the implementation of inits/tails isn't shared between list and non-empty list.  
Is this because of module dependency issues?

Reply:
Indeed, `inits` gets used under `Membership.Propositional` in such a way as to (seem to) greatly complicate the dependency graph if I were to try to factor out a common root definition under `NonEmpty`... but I I will investigate further while undertaking the refactoring requested below.

Reply:
Suggest instead: `inits-tail` and `tails-tail` for the auxiliaries?

Reply:
Ah! Much better... and yes, AFAIK, we would/should/do allow `tail∘inits` and `tail∘tails` (and isn't that better, for uniformity's sake?)